include/nbl/core/memory/memory.h

```diff
@@ -49,7 +49,7 @@ namespace impl
         if (alignment < MIN_ALIGN) alignment = MIN_ALIGN;
         if (size == 0) return nullptr;
         void* p;
-        if (::posix_memalign(&p, alignment>alignof(std::max_align_t) ? alignof(std::max_align_t):alignment, size) != 0) p = nullptr;
```

Comment:
I couldn't figure out why this originally sets an upper bound of 16 here. 

I was looking at the man pages for https://linux.die.net/man/3/posix_memalign the only restriction is that is had to be a power of two.

```
EINVAL

The alignment argument was not a power of two, or was not a multiple of sizeof(void *). 
```

Reply:
ah, `max_align_t` is a constraint on your stack https://en.cppreference.com/w/cpp/types/max_align_t

> Pointers returned by allocation functions such as [std::malloc](https://en.cppreference.com/w/cpp/memory/c/malloc) are suitably aligned for any object, which means they are aligned at least as strictly as std::max_align_t.

Basically its not really valid to declare structs or classes `alignas(A) T` with `A>alignof(std::max_align_t)` or stuff like this will fail:
```cpp
T t;
assert((ptrdiff_t(&t)%alignof(T))==0);
```

But yeah you're right, for the heap it shouldn't matter